src/Development/IDE/Core/Rules.hs

```diff
@@ -642,10 +639,10 @@ typeCheckRuleDefinition hsc pm isFoi source = do
 generateCore :: RunSimplifier -> NormalizedFilePath -> Action (IdeResult (SafeHaskellMode, CgGuts, ModDetails))
 generateCore runSimplifier file = do
     deps <- use_ GetDependencies file
-    (tm:tms) <- uses_ TypeCheck (file:transitiveModuleDeps deps)
+    (tm:_) <- uses_ TypeCheck (file:transitiveModuleDeps deps)
     setPriority priorityGenerateCore
     packageState <- hscEnv <$> use_ GhcSession file
-    liftIO $ compileModule runSimplifier packageState [(tmrModSummary x, tmrModInfo x) | x <- tms] tm
+    liftIO $ compileModule runSimplifier packageState tm
```

Comment:
surprised that this can work with `GhcSession` and doesn't need `GhcSessionDeps`